cbmc/proofs/prepare.py

```diff
@@ -24,16 +24,52 @@
 
 import logging
 import os
+import pathlib
+import subprocess
 import sys
 import textwrap
-from subprocess import CalledProcessError
 
 from make_cbmc_batch_files import create_cbmc_yaml_files
 
+
+def apply_patches():
+    patch_dir = pathlib.Path(__file__).resolve().parent.parent / "patches"
```

Comment:
Does this really work?  Creates a string to a patches directory?

Reply:
It creates a Path object, which returns the path as a string if you wrap it in `str()` or print it. Otherwise, it's more convenient than `os.path` because you can directly call methods on it and access its `parent` member etc.

Reply:
I'm concerned about the / notation.  It appears to be trying to form a string without using a string.

Reply:
I've confirmed it does work, but I don't understand the python mechanism that makes it work.